flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -381,11 +394,44 @@
     // ------------------------------------------------------------------------------------------
 
     /** The enforcer to guarantee NOT NULL column constraint when writing data into sink. */
-    public enum NotNullEnforcer {
-        /** Throws runtime exception when writing null values into NOT NULL column. */
-        ERROR,
-        /** Drop records when writing null values into NOT NULL column. */
-        DROP
+    public enum NotNullEnforcer implements DescribedEnum {
+        ERROR(text("Throws runtime exception when writing null values into NOT NULL column.")),
+        DROP(text("Drops records when writing null values into NOT NULL column."));
+
+        private final InlineElement description;
+
+        NotNullEnforcer(InlineElement description) {
+            this.description = description;
+        }
+
+        @Override
+        public InlineElement getDescription() {
+            return description;
+        }
+    }
+
+    /**
+     * The enforcer to guarantee that precision of CHAR/VARCHAR columns is respected when writing
+     * data into sink.
+     */
+    public enum CharPrecisionEnforcer implements DescribedEnum {
+        IGNORE(
+                text(
+                        "Doesn't apply any trimming, simply ignores the CHAR/VARCHAR precision directive.")),
+        TRIM(
+                text(
+                        "Trims string values to match the length defined as the CHAR/VARCHAR precision."));
```

Comment:
Just for consistent wording

```suggestion
                        "Trim string values to match the length defined by the CHAR/VARCHAR precision."));
```